codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/protocols/parse/EventStreamUnmarshallerGenerator.kt

```diff
@@ -259,36 +262,37 @@ class EventStreamUnmarshallerGenerator(
                 *codegenScope,
             )
         }
-        val memberName = symbolProvider.toMemberName(member)
-        withBlock("builder = builder.$memberName(", ");") {
-            when (target) {
-                is BlobShape -> {
-                    rustTemplate("#{Blob}::new(message.payload().as_ref())", *codegenScope)
-                }
-                is StringShape -> {
-                    rustTemplate(
-                        """
-                        std::str::from_utf8(message.payload())
-                            .map_err(|_| #{Error}::Unmarshalling("message payload is not valid UTF-8".into()))?
-                        """,
-                        *codegenScope,
-                    )
-                }
-                is UnionShape, is StructureShape -> {
-                    renderParseProtocolPayload(member)
+        withBlock("builder = builder.${member.setterName()}(", ");") {
+            conditionalBlock("Some(", ")", member.isOptional) {
+                when (target) {
+                    is BlobShape -> {
+                        rustTemplate("#{Blob}::new(message.payload().as_ref())", *codegenScope)
+                    }
+                    is StringShape -> {
+                        rustTemplate(
+                            """
+                            std::str::from_utf8(message.payload())
+                                .map_err(|_| #{Error}::Unmarshalling("message payload is not valid UTF-8".into()))?
+                                .to_owned()
+                            """,
+                            *codegenScope,
+                        )
+                    }
+                    is UnionShape, is StructureShape -> {
+                        renderParseProtocolPayload(member)
+                    }
                 }
             }
         }
     }
 
     private fun RustWriter.renderParseProtocolPayload(member: MemberShape) {
         val parser = protocol.structuredDataParser(operationShape).payloadParser(member)
-        val memberName = symbolProvider.toMemberName(member)
         rustTemplate(
             """
             #{parser}(&message.payload()[..])
                 .map_err(|err| {
-                    #{Error}::Unmarshalling(format!("failed to unmarshall $memberName: {}", err))
+                    #{Error}::Unmarshalling(format!("failed to unmarshall ${member.memberName}: {}", err))
```

Comment:
We're using `member.memberName` as the discriminator and in error messages pervasively. Union member shape symbol names are just being pascal cased.

I'll make a note here that this needs further investigation; I suspect we may already have inconsistencies of the sort I uncovered in https://github.com/awslabs/smithy-rs/issues/1826 for operation names.

Reply:
This may be a case where the SDK and server want different things. Servers will want to use the modeled name in errors since the people working on them will be familiar with the model, whereas, customers of the SDK don't see and aren't familiar with the model at all, so the Rust name is actually better for them.

Reply:
I've reverted this.